main.py

```diff
@@ -280,14 +283,44 @@ def render_field_link(field):
         name=field['name'],
         id=field_id)
 
+def render_scouting_observation_link(scouting_observation):
+    id = scouting_observation['id']
+    return '<a href="{link}">{id}</a>'.format(
+        link=url_for('scouting_observation', scouting_observation_id=id),
+        id=id)
 
 def redirect_uri():
     """
     :return: Returns uri for redirection after Log In with FieldView.
     """
     return url_for('login_redirect', _external=True)
 
+@app.route('/scouting-observation/<scouting_observation_id>', methods=['GET'])
+def scouting_observation(scouting_observation_id):
+    observation = climate.get_scouting_observation(state('access_token'), CLIMATE_API_KEY, scouting_observation_id)
+   
+    return  """
+            <h1>Partner API Demo Site</h1>
+            <h2>Scouting Observation ID: {scouting_observation_id}</h2>
+            <p>{json}</p>
+            <p><a href='{observations}'>Return to Observations list</a></p>
+            """.format(scouting_observation_id=scouting_observation_id,json=json.dumps(observation), observations=url_for('scouting_observations'))
+
+
+
+@app.route('/scouting-observations', methods=['GET'])
+def scouting_observations():
+    observations = climate.get_scouting_observations(state('access_token'), CLIMATE_API_KEY, 100, None)
+    html = "<h1>Partner API Demo Site</h1>"
+    if observations:
+        scouting_observations = render_ul(render_scouting_observation_link(f) for f in observations)
+        html +="<p>Your Climate Scouting Observations:{scouting_observations}</p>".format(scouting_observations=scouting_observations)
+    else:
+        html+= "<p>No Scouting Observations found!"
+
+    html+= "<p><a href='{home}'>Return home</a></p>".format(home=url_for('home'))
 
+    return html
```

Comment:
how about something like

```py
if observations:
    observations_list = render_ul(render_scouting_observation_link(o) for o in observations)
    body = "Your Climate Scouting Observations: {observations_list}".format(observations_list)
else:
    body = "No Scouting Observations found!"

return """
       <h1>Partner API Demo Site</h1>
       <p>{body}</p>
       <p><a href='{home}'>Return home</a></p>
       """.format(body=body,
                  home=url_for('home'))
```

for me personally, this pattern helps communicate what the HTML will look like without needing to read the code and following the string concatenation.

Reply:
+1 to this

Reply:
Accepted